amount_helper.rb

```
@@ -0,0 +1,54 @@
+module AmountHelper
+  def amount(number, options = {})
+    if number.nil? && options[:nil_as_dash] != false
+      ndash
+    elsif number.zero? && options[:zero_as_dash]
+      ndash
+    elsif number.present?
+      precision = options[:precision]
+      if precision
+        options[:minimum_precision] = precision
+        options[:maximum_precision] = precision
+      end
+
+      number = number.round if options[:round]
+      separator = options.fetch(:separator, I18n.t('number.format.separator'))
+      delimiter = options.fetch(:delimiter, I18n.t('number.format.delimiter')).to_s
+      minimum_precision = options[:minimum_precision] || 0
+      str = number.to_s.sub(/\.0+\z/, '')
+      str =~ /\A(\-?)(\d+)(?:\.(\d+))?\z/ || raise("Could not parse number: #{number}")
```

You should raise this error earlier.